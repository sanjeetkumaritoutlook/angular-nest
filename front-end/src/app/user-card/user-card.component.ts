import { Component, Input, Output, EventEmitter  } from '@angular/core';

@Component({
  selector: 'app-user-card',
  template: `
  <div class="card">
    <h2>{{ user.name }}</h2>
    <p>Email: {{ user.email }}</p>
    <p>Role: {{ user.role }}</p>
    <button (click)="selectUser()">Select</button>
  </div>
`,
styles: [`
  .card { border: 1px solid #ccc; padding: 10px; margin: 10px; }
  h2 { margin: 0; }
`]
})
export class UserCardComponent {
  @Input() user!: { name: string; email: string; role: string };
  @Output() userSelected = new EventEmitter<string>();

  selectUser() {
    this.userSelected.emit(this.user.name);
  }
}
