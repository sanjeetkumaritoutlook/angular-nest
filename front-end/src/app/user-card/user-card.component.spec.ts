import { ComponentFixture, TestBed } from '@angular/core/testing';

import { UserCardComponent } from './user-card.component';
import { By } from '@angular/platform-browser';

describe('UserCardComponent', () => {
  let component: UserCardComponent;
  let fixture: ComponentFixture<UserCardComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [UserCardComponent]
    })
    .compileComponents();
    
    fixture = TestBed.createComponent(UserCardComponent);
    component = fixture.componentInstance;

     // ✅ Provide a mock user object
     component.user = {
      name: 'John Doe',
      role: 'Admin',
      email: 'john.doe@example.com', 
    };

    fixture.detectChanges(); // ✅ Call after setting user
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  //new test cases below:
  it('should match snapshot with default user', () => {
    component.user = { name: 'John Doe', email: 'john@example.com', role: 'Admin' };
    fixture.detectChanges();
    
    expect(fixture.nativeElement.innerHTML).toMatchSnapshot();
  });

  it('should emit event when button is clicked', () => {
    jest.spyOn(component.userSelected, 'emit');
    
    const button = fixture.debugElement.query(By.css('button'));
    button.triggerEventHandler('click', null);
    console.log(component.user);
     // Expect only the user's name to be emitted
   expect(component.userSelected.emit).toHaveBeenCalledWith(component.user.name);

  });
  it('should match snapshot with another user', () => {
    component.user = { name: 'Jane Smith', email: 'jane@example.com', role: 'User' };
    fixture.detectChanges();

    expect(fixture.nativeElement.innerHTML).toMatchSnapshot();
  });

  it('should emit event when button is clicked', () => {
    component.user = { name: 'Alice Brown', email: 'alice@example.com', role: 'Manager' };
    fixture.detectChanges();

    jest.spyOn(component.userSelected, 'emit');
    const button = fixture.debugElement.query(By.css('button')).nativeElement;
    button.click();

    expect(component.userSelected.emit).toHaveBeenCalledWith('Alice Brown');
  });
});
