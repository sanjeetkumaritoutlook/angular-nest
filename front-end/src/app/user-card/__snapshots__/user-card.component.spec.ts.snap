// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCardComponent should match snapshot with another user 1`] = `"<div class="card"><h2>Jane Smith</h2><p>Email: jane@example.com</p><p>Role: User</p><button>Select</button></div>"`;

exports[`UserCardComponent should match snapshot with default user 1`] = `"<div class="card"><h2>John Doe</h2><p>Email: john@example.com</p><p>Role: Admin</p><button>Select</button></div>"`;
